docs/proposals/api-group.md

```diff
@@ -0,0 +1,104 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/api-group.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Supporting multiple API groups
+
+## Goal
+
+1. Breaking the monolithic v1 API to modular groups
+
+2. Supporting different versions in different groups
+
+3. Supporting identically named kinds to exist in different groups
+
+4. Enabling/disabling groups individually
+
+5. Laying the basis for [API Plugin](../../docs/design/extending-api.md)
+
+6. Allowing users to omit group name when using kubectl
+
+
+## Bookkeeping for groups
+
+1. No changes to API spec:
+
+  We will not add new field in the API spec to hold the group information. Group will be part of the TypeMeta.APIVersion field, in the form of `group/version`.
```

Comment:
> I want these values to be opaque to most client code.  The only thing that
> needs to know about group is a RESTMapper and the inner guts of conversion.

I'm fine with attempting to disambiguate intent server-side.  I'm fine attempting to disambiguate intent client side.  But I don't think that either case necessitates glomming fields together in our data.

The only case where I see the distinction matter is when using an old client to retrieve data from a new api group.  In such a case, the client might get confused and expect to find the `kind` in the `version` for our current scheme.  I don't think I care about that case, because it means you've done crazy things to point at something other than `api/v1`.